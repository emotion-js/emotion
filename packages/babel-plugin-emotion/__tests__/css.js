import babelTester from 'babel-tester'
import plugin from 'babel-plugin-emotion'
import path from 'path'

babelTester(
  '@emotion/babel-plugin-core css',
  path.join(__dirname, 'css-macro'),
  {
    plugins: [plugin],
    transform: src => src.replace(/\/macro/g, '')
  }
)
