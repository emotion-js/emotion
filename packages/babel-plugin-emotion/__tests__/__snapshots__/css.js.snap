// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/babel-plugin-core css actual-expected-usage 1`] = `
"/** @jsx jsx */
import { jsx } from '@emotion/core'
import css from '@emotion/css'

const SomeComponent = () => (
  <div
    css={css\`
      color: hotpink;
    \`}
  />
)


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

/** @jsx jsx */
import { jsx } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\"
} : {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImFjdHVhbC1leHBlY3RlZC11c2FnZS5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFNWSIsImZpbGUiOiJhY3R1YWwtZXhwZWN0ZWQtdXNhZ2UuanMiLCJzb3VyY2VzQ29udGVudCI6WyIvKiogQGpzeCBqc3ggKi9cbmltcG9ydCB7IGpzeCB9IGZyb20gJ0BlbW90aW9uL2NvcmUnXG5pbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuY29uc3QgU29tZUNvbXBvbmVudCA9ICgpID0+IChcbiAgPGRpdlxuICAgIGNzcz17Y3NzYFxuICAgICAgY29sb3I6IGhvdHBpbms7XG4gICAgYH1cbiAgLz5cbilcbiJdfQ== */\\"
};

const SomeComponent = () => <div css={_ref} />;"
`;

exports[`@emotion/babel-plugin-core css basic 1`] = `
"import css from '@emotion/css'

function doThing() {
  return css\`
    display: flex;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"k008qs\\",
  styles: \\"display:flex;\\"
} : {
  name: \\"k008qs\\",
  styles: \\"display:flex;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImJhc2ljLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUdZIiwiZmlsZSI6ImJhc2ljLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmZ1bmN0aW9uIGRvVGhpbmcoKSB7XG4gIHJldHVybiBjc3NgXG4gICAgZGlzcGxheTogZmxleDtcbiAgYFxufVxuIl19 */\\"
};

function doThing() {
  return _ref;
}"
`;

exports[`@emotion/babel-plugin-core css call-expression 1`] = `
"import css from '@emotion/css'

function doThing() {
  return css({ color: 'hotpink' })
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGwtZXhwcmVzc2lvbi5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHUyIsImZpbGUiOiJjYWxsLWV4cHJlc3Npb24uanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuZnVuY3Rpb24gZG9UaGluZygpIHtcbiAgcmV0dXJuIGNzcyh7IGNvbG9yOiAnaG90cGluaycgfSlcbn1cbiJdfQ== */\\"
};

function doThing() {
  return _ref;
}"
`;

exports[`@emotion/babel-plugin-core css call-inside-call 1`] = `
"import css from '@emotion/css'

const thing = css\`
  display: flex;
  &:hover {
    \${css\`
      color: hotpink;
    \`};
  }
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

const thing =
/*#__PURE__*/
_css(\\"display:flex;&:hover{\\", process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\"
} : {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGwtaW5zaWRlLWNhbGwuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBS1MiLCJmaWxlIjoiY2FsbC1pbnNpZGUtY2FsbC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5jb25zdCB0aGluZyA9IGNzc2BcbiAgZGlzcGxheTogZmxleDtcbiAgJjpob3ZlciB7XG4gICAgJHtjc3NgXG4gICAgICBjb2xvcjogaG90cGluaztcbiAgICBgfTtcbiAgfVxuYFxuIl19 */\\"
}, \\";}label:thing;\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNhbGwtaW5zaWRlLWNhbGwuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBRWlCIiwiZmlsZSI6ImNhbGwtaW5zaWRlLWNhbGwuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuY29uc3QgdGhpbmcgPSBjc3NgXG4gIGRpc3BsYXk6IGZsZXg7XG4gICY6aG92ZXIge1xuICAgICR7Y3NzYFxuICAgICAgY29sb3I6IGhvdHBpbms7XG4gICAgYH07XG4gIH1cbmBcbiJdfQ== */\\"));"
`;

exports[`@emotion/babel-plugin-core css comment-with-interpolation 1`] = `
"import css from '@emotion/css'

css\`
  // color: \${'green'};
  /* 
  
  something: \${'something'};
  
  */
  color: hotpink;
\`

css\`
  // color: \${'green'};
  /* 
  
  something: \${'something'};
  
  */
  color: \${'hotpink'};
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNvbW1lbnQtd2l0aC1pbnRlcnBvbGF0aW9uLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUVHIiwiZmlsZSI6ImNvbW1lbnQtd2l0aC1pbnRlcnBvbGF0aW9uLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmNzc2BcbiAgLy8gY29sb3I6ICR7J2dyZWVuJ307XG4gIC8qIFxuICBcbiAgc29tZXRoaW5nOiAkeydzb21ldGhpbmcnfTtcbiAgXG4gICovXG4gIGNvbG9yOiBob3RwaW5rO1xuYFxuXG5jc3NgXG4gIC8vIGNvbG9yOiAkeydncmVlbid9O1xuICAvKiBcbiAgXG4gIHNvbWV0aGluZzogJHsnc29tZXRoaW5nJ307XG4gIFxuICAqL1xuICBjb2xvcjogJHsnaG90cGluayd9O1xuYFxuIl19 */\\"
};
process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImNvbW1lbnQtd2l0aC1pbnRlcnBvbGF0aW9uLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQVlHIiwiZmlsZSI6ImNvbW1lbnQtd2l0aC1pbnRlcnBvbGF0aW9uLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmNzc2BcbiAgLy8gY29sb3I6ICR7J2dyZWVuJ307XG4gIC8qIFxuICBcbiAgc29tZXRoaW5nOiAkeydzb21ldGhpbmcnfTtcbiAgXG4gICovXG4gIGNvbG9yOiBob3RwaW5rO1xuYFxuXG5jc3NgXG4gIC8vIGNvbG9yOiAkeydncmVlbid9O1xuICAvKiBcbiAgXG4gIHNvbWV0aGluZzogJHsnc29tZXRoaW5nJ307XG4gIFxuICAqL1xuICBjb2xvcjogJHsnaG90cGluayd9O1xuYFxuIl19 */\\"
};"
`;

exports[`@emotion/babel-plugin-core css impure 1`] = `
"import css from '@emotion/css'

function thing() {}

function doThing() {
  return css\`
    display: \${thing()};
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

function thing() {}

function doThing() {
  return (
    /*#__PURE__*/
    _css(\\"display:\\", thing(), \\";\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImltcHVyZS5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFLWSIsImZpbGUiOiJpbXB1cmUuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuZnVuY3Rpb24gdGhpbmcoKSB7fVxuXG5mdW5jdGlvbiBkb1RoaW5nKCkge1xuICByZXR1cm4gY3NzYFxuICAgIGRpc3BsYXk6ICR7dGhpbmcoKX07XG4gIGBcbn1cbiJdfQ== */\\"))
  );
}"
`;

exports[`@emotion/babel-plugin-core css inside-anonymous-arrow-function 1`] = `
"import css from '@emotion/css'

export default () => {
  css\`
    color: hotpink;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluc2lkZS1hbm9ueW1vdXMtYXJyb3ctZnVuY3Rpb24uanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBR0siLCJmaWxlIjoiaW5zaWRlLWFub255bW91cy1hcnJvdy1mdW5jdGlvbi5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5leHBvcnQgZGVmYXVsdCAoKSA9PiB7XG4gIGNzc2BcbiAgICBjb2xvcjogaG90cGluaztcbiAgYFxufVxuIl19 */\\"
};

export default (() => {
  _ref;
});"
`;

exports[`@emotion/babel-plugin-core css inside-anonymous-function 1`] = `
"import css from '@emotion/css'

export default () => {
  css\`
    color: hotpink;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluc2lkZS1hbm9ueW1vdXMtZnVuY3Rpb24uanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBR0siLCJmaWxlIjoiaW5zaWRlLWFub255bW91cy1mdW5jdGlvbi5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5leHBvcnQgZGVmYXVsdCAoKSA9PiB7XG4gIGNzc2BcbiAgICBjb2xvcjogaG90cGluaztcbiAgYFxufVxuIl19 */\\"
};

export default (() => {
  _ref;
});"
`;

exports[`@emotion/babel-plugin-core css inside-class 1`] = `
"/** @jsx jsx */
import { Component } from 'react'
import { jsx } from '@emotion/core'
import css from '@emotion/css'

class SomeComponent extends Component {
  render() {
    return (
      <div
        css={css\`
          color: hotpink;
        \`}
      />
    )
  }
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

/** @jsx jsx */
import { Component } from 'react';
import { jsx } from '@emotion/core';

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\"
} : {
  name: \\"1wl9j1m-SomeComponent\\",
  styles: \\"color:hotpink;label:SomeComponent;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluc2lkZS1jbGFzcy5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFTZ0IiLCJmaWxlIjoiaW5zaWRlLWNsYXNzLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBDb21wb25lbnQgfSBmcm9tICdyZWFjdCdcbmltcG9ydCB7IGpzeCB9IGZyb20gJ0BlbW90aW9uL2NvcmUnXG5pbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuY2xhc3MgU29tZUNvbXBvbmVudCBleHRlbmRzIENvbXBvbmVudCB7XG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gKFxuICAgICAgPGRpdlxuICAgICAgICBjc3M9e2Nzc2BcbiAgICAgICAgICBjb2xvcjogaG90cGluaztcbiAgICAgICAgYH1cbiAgICAgIC8+XG4gICAgKVxuICB9XG59XG4iXX0= */\\"
};

class SomeComponent extends Component {
  render() {
    return <div css={_ref} />;
  }

}"
`;

exports[`@emotion/babel-plugin-core css inside-non-pascal-case-arrow-function 1`] = `
"import css from '@emotion/css'

let something = () => {
  css\`
    color: hotpink;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImluc2lkZS1ub24tcGFzY2FsLWNhc2UtYXJyb3ctZnVuY3Rpb24uanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBR0siLCJmaWxlIjoiaW5zaWRlLW5vbi1wYXNjYWwtY2FzZS1hcnJvdy1mdW5jdGlvbi5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5sZXQgc29tZXRoaW5nID0gKCkgPT4ge1xuICBjc3NgXG4gICAgY29sb3I6IGhvdHBpbms7XG4gIGBcbn1cbiJdfQ== */\\"
};

let something = () => {
  _ref;
};"
`;

exports[`@emotion/babel-plugin-core css label-1 1`] = `
"import css from '@emotion/css'

const thing = css\`
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
const thing = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\"
} : {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImxhYmVsLTEuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBRWlCIiwiZmlsZSI6ImxhYmVsLTEuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuY29uc3QgdGhpbmcgPSBjc3NgXG4gIGNvbG9yOiBob3RwaW5rO1xuYFxuIl19 */\\"
};"
`;

exports[`@emotion/babel-plugin-core css label-object 1`] = `
"import css from '@emotion/css'

const thing = {
  thisShouldBeTheLabel: css\`
    color: hotpink;
  \`,
  // prettier-ignore
  'shouldBeAnotherLabel':css\`
    color:green;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
const thing = {
  thisShouldBeTheLabel: process.env.NODE_ENV === \\"production\\" ? {
    name: \\"xjpive-thisShouldBeTheLabel\\",
    styles: \\"color:hotpink;label:thisShouldBeTheLabel;\\"
  } : {
    name: \\"xjpive-thisShouldBeTheLabel\\",
    styles: \\"color:hotpink;label:thisShouldBeTheLabel;\\",
    map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImxhYmVsLW9iamVjdC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHMkIiLCJmaWxlIjoibGFiZWwtb2JqZWN0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmNvbnN0IHRoaW5nID0ge1xuICB0aGlzU2hvdWxkQmVUaGVMYWJlbDogY3NzYFxuICAgIGNvbG9yOiBob3RwaW5rO1xuICBgLFxuICAvLyBwcmV0dGllci1pZ25vcmVcbiAgJ3Nob3VsZEJlQW5vdGhlckxhYmVsJzpjc3NgXG4gICAgY29sb3I6Z3JlZW47XG4gIGBcbn1cbiJdfQ== */\\"
  },
  // prettier-ignore
  'shouldBeAnotherLabel': process.env.NODE_ENV === \\"production\\" ? {
    name: \\"lgalrw-shouldBeAnotherLabel\\",
    styles: \\"color:green;label:shouldBeAnotherLabel;\\"
  } : {
    name: \\"lgalrw-shouldBeAnotherLabel\\",
    styles: \\"color:green;label:shouldBeAnotherLabel;\\",
    map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImxhYmVsLW9iamVjdC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFPNEIiLCJmaWxlIjoibGFiZWwtb2JqZWN0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmNvbnN0IHRoaW5nID0ge1xuICB0aGlzU2hvdWxkQmVUaGVMYWJlbDogY3NzYFxuICAgIGNvbG9yOiBob3RwaW5rO1xuICBgLFxuICAvLyBwcmV0dGllci1pZ25vcmVcbiAgJ3Nob3VsZEJlQW5vdGhlckxhYmVsJzpjc3NgXG4gICAgY29sb3I6Z3JlZW47XG4gIGBcbn1cbiJdfQ== */\\"
  }
};"
`;

exports[`@emotion/babel-plugin-core css multiple-calls 1`] = `
"import css from '@emotion/css'

const thing = css\`
  color: hotpink;
\`

const otherThing = css\`
  color: green;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
const thing = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\"
} : {
  name: \\"1hi0qos-thing\\",
  styles: \\"color:hotpink;label:thing;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm11bHRpcGxlLWNhbGxzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUVpQiIsImZpbGUiOiJtdWx0aXBsZS1jYWxscy5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5jb25zdCB0aGluZyA9IGNzc2BcbiAgY29sb3I6IGhvdHBpbms7XG5gXG5cbmNvbnN0IG90aGVyVGhpbmcgPSBjc3NgXG4gIGNvbG9yOiBncmVlbjtcbmBcbiJdfQ== */\\"
};
const otherThing = process.env.NODE_ENV === \\"production\\" ? {
  name: \\"onvdw0-otherThing\\",
  styles: \\"color:green;label:otherThing;\\"
} : {
  name: \\"onvdw0-otherThing\\",
  styles: \\"color:green;label:otherThing;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm11bHRpcGxlLWNhbGxzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQU1zQiIsImZpbGUiOiJtdWx0aXBsZS1jYWxscy5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBjc3MgZnJvbSAnQGVtb3Rpb24vY3NzJ1xuXG5jb25zdCB0aGluZyA9IGNzc2BcbiAgY29sb3I6IGhvdHBpbms7XG5gXG5cbmNvbnN0IG90aGVyVGhpbmcgPSBjc3NgXG4gIGNvbG9yOiBncmVlbjtcbmBcbiJdfQ== */\\"
};"
`;

exports[`@emotion/babel-plugin-core css no-actual-import 1`] = `
"import '@emotion/css'


      ↓ ↓ ↓ ↓ ↓ ↓

import '@emotion/css';"
`;

exports[`@emotion/babel-plugin-core css other-imports 1`] = `
"import { nonExistantImport } from '@emotion/css'

nonExistantImport()


      ↓ ↓ ↓ ↓ ↓ ↓

import { nonExistantImport as _nonExistantImport } from \\"@emotion/css\\";

_nonExistantImport();"
`;

exports[`@emotion/babel-plugin-core css remove-block-comments 1`] = `
"import css from '@emotion/css'

css\`
  /* color:green;
ddjfwjkng
  */
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlbW92ZS1ibG9jay1jb21tZW50cy5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFFRyIsImZpbGUiOiJyZW1vdmUtYmxvY2stY29tbWVudHMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuY3NzYFxuICAvKiBjb2xvcjpncmVlbjtcbmRkamZ3amtuZ1xuICAqL1xuICBjb2xvcjogaG90cGluaztcbmBcbiJdfQ== */\\"
};"
`;

exports[`@emotion/babel-plugin-core css remove-line-comments 1`] = `
"import css from '@emotion/css'

css\`
  // color: green;
  color: hotpink;
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";
process.env.NODE_ENV === \\"production\\" ? {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\"
} : {
  name: \\"1lrxbo5\\",
  styles: \\"color:hotpink;\\",
  map: \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlbW92ZS1saW5lLWNvbW1lbnRzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUVHIiwiZmlsZSI6InJlbW92ZS1saW5lLWNvbW1lbnRzLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmNzc2BcbiAgLy8gY29sb3I6IGdyZWVuO1xuICBjb2xvcjogaG90cGluaztcbmBcbiJdfQ== */\\"
};"
`;

exports[`@emotion/babel-plugin-core css tagged-template-args-forwarded 1`] = `
"import css from '@emotion/css'

function media(...args) {
  return css\`
    @media (min-width: 100px) {
      \${css(...args)};
    }
  \`
}

const test = css\`
  \${media\`color: red;\`};
\`


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

function media(...args) {
  return (
    /*#__PURE__*/
    _css(\\"@media (min-width:100px){\\",
    /*#__PURE__*/
    _css(...args), \\";}\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRhZ2dlZC10ZW1wbGF0ZS1hcmdzLWZvcndhcmRlZC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHWSIsImZpbGUiOiJ0YWdnZWQtdGVtcGxhdGUtYXJncy1mb3J3YXJkZWQuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgY3NzIGZyb20gJ0BlbW90aW9uL2NzcydcblxuZnVuY3Rpb24gbWVkaWEoLi4uYXJncykge1xuICByZXR1cm4gY3NzYFxuICAgIEBtZWRpYSAobWluLXdpZHRoOiAxMDBweCkge1xuICAgICAgJHtjc3MoLi4uYXJncyl9O1xuICAgIH1cbiAgYFxufVxuXG5jb25zdCB0ZXN0ID0gY3NzYFxuICAke21lZGlhYGNvbG9yOiByZWQ7YH07XG5gXG4iXX0= */\\"))
  );
}

const test =
/*#__PURE__*/
_css(media\`color: red;\`, \\";label:test;\\" + (process.env.NODE_ENV === \\"production\\" ? \\"\\" : \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRhZ2dlZC10ZW1wbGF0ZS1hcmdzLWZvcndhcmRlZC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFVZ0IiLCJmaWxlIjoidGFnZ2VkLXRlbXBsYXRlLWFyZ3MtZm9yd2FyZGVkLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGNzcyBmcm9tICdAZW1vdGlvbi9jc3MnXG5cbmZ1bmN0aW9uIG1lZGlhKC4uLmFyZ3MpIHtcbiAgcmV0dXJuIGNzc2BcbiAgICBAbWVkaWEgKG1pbi13aWR0aDogMTAwcHgpIHtcbiAgICAgICR7Y3NzKC4uLmFyZ3MpfTtcbiAgICB9XG4gIGBcbn1cblxuY29uc3QgdGVzdCA9IGNzc2BcbiAgJHttZWRpYWBjb2xvcjogcmVkO2B9O1xuYFxuIl19 */\\"));"
`;
